src/index.ts

```diff
@@ -225,56 +225,64 @@ class Egnyte {
    */
-  async getAvailableLicensing (): Promise<any> {
-    const authCookie = await this._authenticate(this._config.username, this._config.password)
+  async getPlans(): Promise<any> {
+    const authCookie = await this._authenticate()
     const planIds = await this._getAllPlanIds(authCookie)
 
-    const promises = []
+    const results = await Promise.all(
+      planIds.map(async id => {
+        const [planDataRes, planPowerUserDataRes] = await Promise.all([
+          this._egnyteRequest(`/msp/usage_stats/${this.resellerId}/${id}/`, {
+            headers: { cookie: authCookie },
+          }),
+          this._egnyteRequest(`/msp/get_plan_pu_data/${this.resellerId}/${id}/`, {
+            headers: { cookie: authCookie },
+          }),
+        ])
 
-    for (const id of planIds) {
-      promises.push(got(`https://resellers.egnyte.com/msp/usage_stats/${this._config.resellerId}/${id}/`, {
-        ...this._gotConfigBase,
-        headers: { cookie: authCookie },
-        responseType: 'json'
-      }))
-    }
+        const planData = planDataRes.data
+        const planPowerUserData = planPowerUserDataRes.data
 
-    const result: any[] = await Promise.all(promises)
+        const ref: any = Object.entries(planData[0])[0][1]
 
-    const final = []
-    for (const i in result) {
-      const base = result[i].body[0]
-      const arr: any[] = Object.entries(base)[0]
-      const data = arr[1]
-      final.push({
-        planId: planIds[i],
-        availablePowerUsers: data.power_user_stats.Available,
-        availableStorage: data.storage_stats.Available
-      })
-    }
+        return {
+          planId: id,
+          totalPowerUsers: planPowerUserData.purchased,
+          usedPowerUsers: planPowerUserData.purchased - ref.power_user_stats.Available,
+          availablePowerUsers: ref.power_user_stats.Available,
+          availableStorage: ref.storage_stats.Available,
+          customers: planData.map((customer: object) => Object.keys(customer)[0]),
+        }
+      }),
+    )
 
-    return final
+    return results
   }
 
-  async UpdatePowerUserLicensing (planId: string, newTotalLicenses: number) {
-    const authCookie = await this._authenticate(this._config.username, this._config.password)
-    const res: any = await got(`https://resellers.egnyte.com/msp/change_plan_power_users/${this._config.resellerId}/`, {
-      ...this._gotConfigBase,
-      method: 'post',
-      headers: {
-        Cookie: authCookie,
-        'Content-Type': 'application/json',
-        'X-Requested-With': 'XMLHttpRequest'
+  /**
+   * Updates a plan's power user licensing count. This will result in a billing change from egnyte. Must be increased in increments of 5
+   * @param planId the planId to update licensing for
+   * @param newTotalLicenses the new total in increments of 5. Use getPlans() to find current total for given plan
+   */
+  async UpdatePowerUserLicensing(planId: string, newTotalLicenses: number) {
```

Comment:
Does Egnyte toss a friendly error upstream if the `newTotalLicenses` number isn't a multiple of 5?

If so, probably easiest to leave as-is. If not, we could potentially add a type guard to check `newTotalLicenses % 5`

Reply:
yeah it errors from service. I felt like the description in IDE would be useful enough but feel free to improve things guys.

Reply:
As long as it errors from the service intelligently, I don't think it's worth the effort. Was only thinking in the event it was some really obscure error that required reverse engineering to figure out.